tests/integration/modifiers/did-intersect-test.js

```diff
@@ -3,85 +3,65 @@ import { setupRenderingTest } from 'ember-qunit';
 import { render } from '@ember/test-helpers';
 import hbs from 'htmlbars-inline-precompile';
 import sinon from 'sinon';
-
-let intersectionCallback;
-let intersectionOptions;
-let observeStub;
-let disconnectStub;
-let MockIntersectionObserver;
+import Service from '@ember/service';
 
 module('Integration | Modifier | did-intersect', function (hooks) {
   setupRenderingTest(hooks);
 
-  let intersectionObserver;
-
   hooks.beforeEach(function () {
-    intersectionCallback = null;
-    intersectionOptions = null;
-    observeStub = sinon.stub();
-    disconnectStub = sinon.stub();
-
-    MockIntersectionObserver = class MockIntersectionObserver {
-      constructor(callback, options) {
-        intersectionCallback = callback;
-        intersectionOptions = options;
+    class MockObserverManager extends Service {
+      constructor() {
+        super(...arguments);
+        this._admin = {};
       }
 
-      observe = observeStub;
-      disconnect = disconnectStub;
+      observe = sinon.stub();
+      unobserve = sinon.stub();
+      addEnterCallback = sinon.stub();
+      addExitCallback = sinon.stub();
     }
 
-    intersectionObserver = window.IntersectionObserver;
-    window.IntersectionObserver = MockIntersectionObserver;
-
-    this.intersectStub = sinon.stub();
-  });
-
-  hooks.afterEach(function () {
-    window.IntersectionObserver = intersectionObserver;
-  });
-
-  test('modifier integrates with IntersectionObserver', async function (assert) {
-    await render(hbs`<div {{did-intersect this.intersectStub}}></div>`);
+    this.owner.register('service:ember-scroll-modifiers@observer-manager', MockObserverManager);
 
-    assert.ok(intersectionCallback, 'IntersectionObserver received callback');
-    assert.ok(observeStub.calledOnce, 'observe was called');
-    assert.ok(intersectionOptions, 'options passed to IntersectionObserver');
-    assert.equal(typeof intersectionOptions, 'object', 'default options is an object');
-    assert.equal(Object.keys(intersectionOptions).length, 0, 'default options is an empty object');
-
-    let [element] = observeStub.args[0];
-
-    assert.ok(element, 'element was passed to observe');
+    this.observerManagerMock = this.owner.lookup('service:ember-scroll-modifiers@observer-manager');
+    this.enterStub = sinon.stub();
+    this.exitStub = sinon.stub();
   });
 
-  test('modifier triggers handler when IntersectionObserver fires callback', async function (assert) {
-    await render(hbs`<div {{did-intersect this.intersectStub}}></div>`);
+  test('modifier integrates with observer-manager', async function (assert) {
```

Comment:
Great rewriting of this test. Can you add three more small tests here?

1) Test that observer and addEnterCallback are still triggered (but addExitCallback is not) if onExit isn't provided.
2) The same thing, but for onEnter instead.
3) Test that the assert is fired if neither of provided. You can do this with the setupOnError test helper to catch the error only for that test.